src/controllers/likes.controller.js

```diff
@@ -0,0 +1,183 @@
+        return util.send(res);
+      }
+      if (!req.body.likeId) {
+        const prof = {
+          userId: req.body.userId,
+          ArticleSlug: req.params.Article,
+          status: 'dislike',
+          claps: 0,
+        };
+        const like = await likeService.createLikes(prof);
+        const newLike = {
+          status: like.status,
+          claps: like.claps,
+          ArticleSlug: like.ArticleSlug,
+        };
+        util.setSuccess(200, 'Successfully disliked', newLike);
+        return util.send(res);
+      }
+      const prof = {
+        status: 'dislike',
+        claps: 0,
+      };
+      if (req.body.status === 'dislike') {
+        prof.status = 'neutral';
+      }
+      const like = await likeService.updateLikes(prof, req.body.likeId);
+      const newLike = {
+        status: like[1].status,
+        claps: like[1].claps,
+        ArticleSlug: like[1].ArticleSlug,
+      };
+      util.setSuccess(200, 'Successfully disliked', newLike);
+      return util.send(res);
+    } catch (error) {
+      util.setError(500, 'server error contact admin');
+      return util.send(res);
+    }
+  }
+
+  /**
+   * @param  {object} req - Request object
+   * @param {object} res - Response object
+   * @returns {object} response
+   *  @static
+   */
+  static async clap(req, res) {
+    try {
+      if (req.body.author === req.body.userId) {
+        util.setError(401, 'You can not clap to your own post');
+        return util.send(res);
+      }
+      if (req.body.likeId) {
+        const prof = {
+          status: 'like',
+          claps: req.body.claps + 1,
+        };
+        const like = await likeService.updateLikes(prof, req.body.likeId);
+        const newLike = {
+          status: like[1].status,
+          claps: like[1].claps,
+          ArticleSlug: like[1].ArticleSlug,
+        };
+        util.setSuccess(200, 'Successfully claped', newLike);
+        return util.send(res);
+      }
+      const prof = {
+        userId: req.body.userId,
+        ArticleSlug: req.params.Article,
+        status: 'like',
+        claps: 1,
+      };
+      const like = await likeService.createLikes(prof);
+      const newLike = {
+        status: like.status,
+        claps: like.claps,
+        ArticleSlug: like.ArticleSlug,
+      };
+      util.setSuccess(200, 'Successfully claped', newLike);
+      return util.send(res);
+    } catch (error) {
+      util.setError(500, 'server error contact admin');
+      return util.send(res);
+    }
+  }
+
+  /**
+   * @param  {object} req - Request object
+   * @param {object} res - Response object
+   * @returns {object} response
+   *  @static
+   */
+  static async getDislikes(req, res) {
+    try {
+      const dislike = await likeService.getAllADislike(req.params.Article);
+      if (dislike) {
+        const data = {
+          dislikes: dislike.count,
+        };
+        util.setSuccess(200, 'Successfully dislike retrieved', data);
```

Comment:
Dislike retrieved successfully would be a better message

Reply:
fixed